tests/test_sshfsmount.cpp

```diff
@@ -73,11 +76,20 @@ struct SshfsMount : public mp::test::SftpServerTest
                                      CommandVector::const_iterator& next_expected_cmd, std::string& output,
                                      bool& invoked)
     {
-        auto request_exec = [&commands, &remaining, &next_expected_cmd, &output, &invoked](ssh_channel,
-                                                                                           const char* raw_cmd) {
+        auto request_exec = [this, &commands, &remaining, &next_expected_cmd, &output, &invoked](ssh_channel,
+                                                                                                 const char* raw_cmd) {
+            invoked = false;
+
             std::string cmd{raw_cmd};
 
-            if (next_expected_cmd != commands.end())
+            auto it = default_cmds.find(cmd);
+            if (it != default_cmds.end())
+            {
+                output = it->second;
+                remaining = output.size();
+                invoked = true;
+            }
+            else if (next_expected_cmd != commands.end())
```

Comment:
This should be fall-through, not the first case. Otherwise you can't test the case when one of the default commands returns something different than the default.

Reply:
Sorry, I'm not really following what you mean here.

Reply:
The defaults should be evaluated last, not first. Otherwise you can't have a `id -u` that returns "1001" in your `commands`.

Reply:
Ah, right, ok, I follow now :smiley:

Reply:
Ah, ok, I follow you now :smiley:

Reply:
Fixed